File: dotnet/Atlas.Averbacao/Services/Retencao/RetencaoService.cs
using Atlas.Averbacao.Api.Retencao;

namespace Atlas.Averbacao.Services.Retencao;

/**
 * Modo Defesa: o servidor pode ser retido. Use request.OportunidadeId
 * e request.ExpiraEm para disparar a sua campanha / contraoferta.
 *
 * Sem Modo Defesa, deixe o método como está.
 */
public class RetencaoService : IRetencaoService
{
    public Task OportunidadeAberta(RetencaoOportunidadeRequest request)
    {
        // await suaApi.AbrirRetencao(request.OportunidadeId, request.ExpiraEm);
        _ = request;
        return Task.CompletedTask;
    }
}

File: java/src/main/java/br/com/atlas/averbacao/services/retencao/RetencaoService.java
package br.com.atlas.averbacao.services.retencao;

import br.com.atlas.averbacao.api.retencao.RetencaoOportunidadeRequest;
import org.springframework.stereotype.Service;

/**
 * Modo Defesa: o servidor pode ser retido. Use request.oportunidadeId()
 * e request.expiraEm() para disparar a sua campanha / contraoferta.
 *
 * Sem Modo Defesa, deixe o método como está.
 */
@Service
public class RetencaoService implements RetencaoContract {
    @Override
    public void oportunidadeAberta(RetencaoOportunidadeRequest request) {
        // suaApi.abrirRetencao(request.oportunidadeId(), request.expiraEm());
    }
}
